transport_test.go

```diff
@@ -133,3 +136,51 @@ func TestTransport_Send(t *testing.T) {
 	// assert send ordering. Sort both slices to be tolerant of re-ordering.
 	require.ElementsMatch(t, expected, received)
 }
+
+type testCountingWriter struct {
+	t *testing.T
+	numCalls *int32
+}
+
+func (tw testCountingWriter) Write(p []byte) (n int, err error) {
+	atomic.AddInt32(tw.numCalls, 1)
+	tw.t.Log("countingWriter", string(p))
+	return len(p), nil
+}
+
+func TestTransport_TcpListenBackoff(t *testing.T) {
```

Comment:
test looks good but I would add more comments around the assertion or a block comment about the methodology (along the lines that you are verifying that there are an expected number of retries given the current max delay / base delay values)

Reply:
Will do. Thanks!